sstables/sstable_set.cc

```diff
@@ -899,7 +899,7 @@ sstable_set_impl::create_single_key_sstable_reader(
     auto readers = boost::copy_range<std::vector<flat_mutation_reader_v2>>(
         filter_sstable_for_reader_by_ck(std::move(selected_sstables), *cf, schema, slice)
         | boost::adaptors::transformed([&] (const shared_sstable& sstable) {
-            tracing::trace(trace_state, "Reading key {} from sstable {}", pos, seastar::value_of([&sstable] { return sstable->get_filename(); }));
+            tracing::trace(trace_state, "Reading key {} from sstable {:D}", pos, *sstable);
```

Comment:
@tchaikov please see if you can make tracing::trace() preserve compile-time format string checking.

Reply:
> @tchaikov please see if you can make tracing::trace() preserve compile-time format string checking.

ack. looking.

Reply:
> I wonder how many bugs this will uncover.

none. see #14169.